app/views/components/_navigation-sidebar.html.erb

```diff
@@ -4,52 +4,70 @@
   publishers ||= []
   collections ||= []
   policies ||= []
-  external_links ||= []
+  other ||= []
   related_content = [
     {"related_items" => related_items},
     {"topics" => topics},
     {"publishers" => publishers},
     {"collections" => collections},
     {"policies" => policies},
-    {"external_links" => external_links}
   ]
 %>
-<% if related_content.map(&:values).flatten.any? %>
+<% if related_content.map(&:values).flatten.any? || other.any? %>
   <aside class="app-c-navigation-sidebar" role="complementary">
     <h2 id="nav-sidebar-related-content" class="app-c-navigation-sidebar__main-heading" >
       <%= t('components.navigation_sidebar.related_content') %>
     </h2>
     <% related_content.each do |section| %>
       <% section.each do |section_title, links| %>
+
         <% if section_title === "related_items" && links.any? %>
           <nav role="navigation" class="app-c-navigation-sidebar__nav-section" aria-labelledby="nav-sidebar-related-content" %>
             <ul class="app-c-navigation-sidebar__link-list">
               <% links.each do |link| %>
                 <li class="app-c-navigation-sidebar__link">
-                  <%= sidebar_link_to(link, section_title) %>
+                  <%= link_to link[:text], link[:path], class: "app-c-navigation-sidebar__related-link", rel: link[:rel] %>
                 </li>
               <% end %>
             </ul>
           </nav>
 
         <% elsif links.any? %>
           <h3 id="nav-sidebar-<%= section_title %>"
-              class="app-c-navigation-sidebar__sub-heading
-              <%= 'app-c-navigation-sidebar__external-heading' if section_title == "external_links" %>">
+              class="app-c-navigation-sidebar__sub-heading">
             <%= t('components.navigation_sidebar.' + section_title) %>
           </h3>
           <nav role="navigation" class="app-c-navigation-sidebar__nav-section" aria-labelledby="nav-sidebar-<%= section_title %>">
             <ul class="app-c-navigation-sidebar__link-list">
               <% links.each do |link| %>
                 <li class="app-c-navigation-sidebar__link">
-                  <%= sidebar_link_to(link, section_title) %>
+                  <%= link_to link[:text], link[:path], class: "app-c-navigation-sidebar__section-link", rel: link[:rel] %>
                 </li>
               <% end %>
             </ul>
           </nav>
+        <% end %>
+      <% end %>
+    <% end %>
 
+    <% if other.any? %>
+      <% other.each do |other_sections| %>
+        <% other_sections.each do |section| %>
+          <h3 class="app-c-navigation-sidebar__sub-heading app-c-navigation-sidebar__sub-heading--bold" id="nav-sidebar-<%= section[:title].parameterize.underscore %>">
+            <%= section[:title] %>
+          </h3>
+          <nav role="navigation" class="app-c-navigation-sidebar__nav-section" aria-labelledby="nav-sidebar-<%= section[:title].parameterize.underscore %>">
```

Comment:
It seems like the same set of elements are repeated for all 3 types of sections and their links.
Would it be possible to reduce this into a partial or helper method to reduce the repetition? I can see there's a bit of variation in css class names so maybe this would be tricky or overly-complex.
Also `other` links aren't the same data structure as the others. Would it be easier to write common code for all 3 types if the data was coerced into the same format before rendering?

Reply:
@steventux This was my first stab at getting something as common as possible - I think the different styling definitely complicates things a bit but I'll have a look today and see if I can simplify it down further.